src/key.rs

```diff
@@ -959,8 +959,8 @@ impl Keypair {
     /// Constructs an schnorr signature for `msg` using the global [`SECP256K1`] context.
     #[inline]
     #[cfg(all(feature = "global-context", feature = "rand-std"))]
-    pub fn sign_schnorr(&self, msg: Message) -> schnorr::Signature {
-        SECP256K1.sign_schnorr(&msg, self)
+    pub fn sign_schnorr(&self, msg: &[u8]) -> schnorr::Signature {
```

Comment:
I'm not really sure about this. On one hand `Message` worked like a lint on the other this API makes much more sense in the context of schnorr sigs. I'm tentatively ACK on this but I'll take more time to think about it and possibly hear other opinions.

Other than that, the PR looks OK.

Reply:
`Message` forced the user to pre-hash their message, which is not required by the upstream API, has no security benefit, and cannot be worked around (it's not just a lint, it's a meaningful API restriction).

Reply:
Please publish this change soon. We'd rather not have to fork the crate to support BIP-340 signatures.
@ChristopherA